Update Tailscale extension

harupong opened the pull request:
fix #18258

## Description

Use IPv4 address to set exit node instead of hostname, so that shared exit node can be selected.

## Screencast

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used in the `README` are located outside the metadata folder if they were not generated with our [metadata tool](https://developers.raycast.com/basics/prepare-an-extension-for-store#how-to-use-it)

ridemountainpig reviewed (changes requested):
Please add an entry to the CHANGELOG.md file. Here are the guidelines: [Version History](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history).

harupong commented:
@ridemountainpig Thanks for the headsup! CHANGELOG.md is updated.

itsmatteomanf reviewed (changes requested):
If we are going the IP route vs. the hostname route (makes sense given the issue presented), we do need to consider that this ACL option exists.

https://tailscale.com/kb/1337/acl-syntax#disableipv4

I'd probably use the IPv6 address, instead of the IPv4 one.
Can't really check right now, but do validate the command works...

harupong commented:
@itsmatteomanf Thanks for taking a look. I agree that IPv6 should be used, not IPv4.

I've modified the code to use IPv6, and validated that the command works as expected.

itsmatteomanf commented:
@harupong sorry for doing this to you (and sorry for the delay, didn't see the notification)... but I just checked, and using the full dns name (with the full address) works. What would you say to do a final update and use that? I'm not sure I can do the fix myself... will approve after.

https://github.com/raycast/extensions/blob/747e981af42cc7bf99bd6a95cdd23c5ee127d9f9/extensions/tailscale/src/shared.tsx#L163

This would result in using, for example, `no-osl-wg-008.mullvad.ts.net.` for one of the Mullvad nodes.

harupong commented:
@itsmatteomanf No problem!! I haven’t done these kind of things for quite some time, so I’m really having fun interacting with the community :)

I modified the related code, and squashed the related commits into one commit. Hope it’s not an issue…

itsmatteomanf reviewed (approved):
Looks good to me!

itsmatteomanf commented:
I think this is ready to be approved and merged, I think we need just @ridemountainpig to remove the requested changes and approve.

Well done @harupong!

ridemountainpig reviewed (approved):
Looks good to me, thanks!
